test/components/views/messages/MBeaconBody-test.tsx

```diff
@@ -333,4 +340,111 @@ describe('<MBeaconBody />', () => {
+        );
+        location2.event.event_id = '2';
+
+        const redactionEvent = new MatrixEvent({ type: EventType.RoomRedaction, content: { reason: 'test reason' } });
+
+        const setupRoomWithBeacon = (locationEvents: MatrixEvent[] = []) => {
+            const room = makeRoomWithStateEvents([aliceBeaconInfo], { roomId, mockClient });
+            const beaconInstance = room.currentState.beacons.get(getBeaconInfoIdentifier(aliceBeaconInfo));
+            beaconInstance.addLocations(locationEvents);
+        };
+        const mockGetRelationsForEvent = (locationEvents: MatrixEvent[] = []) => {
+            const relations = new Relations(RelationType.Reference, M_BEACON.name, mockClient);
+            jest.spyOn(relations, 'getRelations').mockReturnValue(locationEvents);
+
+            const getRelationsForEvent = jest.fn().mockReturnValue(relations);
+
+            return getRelationsForEvent;
+        };
+
+        it('does nothing when getRelationsForEvent is falsy', () => {
+            setupRoomWithBeacon([location1, location2]);
+
+            getComponent({ mxEvent: aliceBeaconInfo });
+
+            act(() => {
+                aliceBeaconInfo.emit(MatrixEventEvent.BeforeRedaction, aliceBeaconInfo, redactionEvent);
+            });
+
+            // no error, no redactions
+            expect(mockClient.redactEvent).not.toHaveBeenCalled();
+        });
+
+        it('cleans up redaction listener on unmount', () => {
+            setupRoomWithBeacon([location1, location2]);
+            const removeListenerSpy = jest.spyOn(aliceBeaconInfo, 'removeListener');
+
+            const component = getComponent({ mxEvent: aliceBeaconInfo });
+
+            act(() => {
+                component.unmount();
+            });
+
+            expect(removeListenerSpy).toHaveBeenCalled();
+        });
+
+        it('does nothing when beacon has no related locations', async () => {
+            // no locations
+            setupRoomWithBeacon();
+            const getRelationsForEvent = await mockGetRelationsForEvent();
+
+            getComponent({ mxEvent: aliceBeaconInfo, getRelationsForEvent });
+
+            act(() => {
+                aliceBeaconInfo.emit(MatrixEventEvent.BeforeRedaction, aliceBeaconInfo, redactionEvent);
+            });
+
+            expect(getRelationsForEvent).toHaveBeenCalledWith(
+                aliceBeaconInfo.getId(), RelationType.Reference, M_BEACON.name,
+            );
+            expect(mockClient.redactEvent).not.toHaveBeenCalled();
+        });
+
+        it('redacts related locations on beacon redaction', async () => {
+            setupRoomWithBeacon([location1, location2]);
+            const getRelationsForEvent = await mockGetRelationsForEvent([location1, location2]);
+
+            getComponent({ mxEvent: aliceBeaconInfo, getRelationsForEvent });
+
+            act(() => {
+                aliceBeaconInfo.emit(MatrixEventEvent.BeforeRedaction, aliceBeaconInfo, redactionEvent);
```

Comment:
Is there any way we can "really" redact an event, instead of emitting a `BeforeRedaction` in at least one test? If the answer is no, fair enough :-)

Reply:
Moved a bit closer to 'really' redacting